exchanges/binance/binance_types.go

```diff
@@ -17,12 +18,1018 @@ const (
+type FuturesOrderGetData struct {
+	AvgPrice      float64 `json:"avgPrice,string"`
+	ClientOrderID string  `json:"clientOrderID"`
+	CumQty        float64 `json:"cumQty,string"`
+	CumBase       float64 `json:"cumBase,string"`
+	ExecutedQty   float64 `json:"executedQty,string"`
+	OrderID       int64   `json:"orderId"`
+	OrigQty       float64 `json:"origQty,string"`
+	OrigType      string  `json:"origType"`
+	Price         float64 `json:"price,string"`
+	ReduceOnly    bool    `json:"reduceOnly"`
+	Side          string  `json:"buy"`
+	PositionSide  string  `json:"positionSide"`
+	Status        string  `json:"status"`
+	StopPrice     float64 `json:"stopPrice,string"`
+	ClosePosition bool    `json:"closePosition"`
+	Symbol        string  `json:"symbol"`
+	Pair          string  `json:"pair"`
+	TimeInForce   string  `json:"timeInForce"`
+	OrderType     string  `json:"type"`
+	ActivatePrice float64 `json:"activatePrice,string"`
+	PriceRate     float64 `json:"priceRate,string"`
+	UpdateTime    int64   `json:"updateTime"`
+	WorkingType   string  `json:"workingType"`
+	PriceProtect  bool    `json:"priceProtect"`
+}
+
+// FuturesOrderData stores order data for futures
+type FuturesOrderData struct {
+	AvgPrice      float64 `json:"avgPrice,string"`
+	ClientOrderID string  `json:"clientOrderId"`
+	CumBase       string  `json:"cumBase"`
+	ExecutedQty   float64 `json:"executedQty,string"`
+	OrderID       int64   `json:"orderId"`
+	OrigQty       float64 `json:"origQty,string"`
+	OrigType      string  `json:"origType"`
+	Price         float64 `json:"price,string"`
+	ReduceOnly    bool    `json:"reduceOnly"`
+	Side          string  `json:"side"`
+	PositionSide  string  `json:"positionSide"`
+	Status        string  `json:"status"`
+	StopPrice     float64 `json:"stopPrice,string"`
+	ClosePosition bool    `json:"closePosition"`
+	Symbol        string  `json:"symbol"`
+	Pair          string  `json:"pair"`
+	Time          int64   `json:"time"`
+	TimeInForce   string  `json:"timeInForce"`
+	OrderType     string  `json:"type"`
+	ActivatePrice float64 `json:"activatePrice,string"`
+	PriceRate     float64 `json:"priceRate,string"`
+	UpdateTime    int64   `json:"updateTime"`
+	WorkingType   string  `json:"workingType"`
+	PriceProtect  bool    `json:"priceProtect"`
+}
+
+// OrderVars stores side, status and type for any order/trade
+type OrderVars struct {
+	Side      order.Side
+	Status    order.Status
+	OrderType order.Type
+	Fee       float64
+}
+
+// AutoCancelAllOrdersData gives data of auto cancelling all open orders
+type AutoCancelAllOrdersData struct {
+	Symbol        string `json:"symbol"`
+	CountdownTime int64  `json:"countdownTime,string"`
+}
+
+// LevelDetail stores level detail data
+type LevelDetail struct {
+	Level         string  `json:"level"`
+	MaxBorrowable float64 `json:"maxBorrowable,string"`
+	InterestRate  float64 `json:"interestRate,string"`
+}
+
+// MarginInfoData stores margin info data
+type MarginInfoData struct {
```

Comment:
this is isolated margin only...
there's another one for cross margin:
https://www.binance.com/gateway-api/v1/friendly/margin/vip/spec/list-all